### PR TITLE
feat(ui-common): integrate `GlideDataGrid` into `MatrixGrid`

### PR DESCRIPTION
**Pull request by hdinia**

### CODE REVIEW

File: webapp/vite.config.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ export default defineConfig(({ mode }) => {
       },
     },
     test: {
-      globals: true, // Use the APIs globally
+      globals: true, // Use the APIs globally,
```
Comment (skamril) [inline webapp/vite.config.ts:33]: ?

File: webapp/src/components/common/MatrixGrid/Matrix.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    redo,
+    canUndo,
+    canRedo,
+  } = useMatrix(study.id, url, enableTimeSeriesColumns, enableAggregateColumns);
+
+  ////////////////////////////////////////////////////////////////
+  // JSX
+  ////////////////////////////////////////////////////////////////
+
+  if (isLoading) {
```
Comment (skamril) [inline webapp/src/components/common/MatrixGrid/Matrix.tsx:52]: Use `UsePromiseCond` component

**Review by skamril — DISMISSED** [review]

File: webapp/vite.config.ts (hunk 1 above)
Comment (hdinia) [reply webapp/vite.config.ts:33]: with that you can use Vitest methods globally see: https://vitest.dev/config/#globals

Comment (hdinia) [reply webapp/vite.config.ts:33]: you don't have to import like vi, test, describe etc each time

**Review by skamril — APPROVED** [review]

File: webapp/src/components/common/MatrixGrid/Matrix.tsx (hunk 2 above)
Comment (hdinia) [reply webapp/src/components/common/MatrixGrid/Matrix.tsx:52]: I prefer not to use it in that case, I handled the errors and loading states

**Pull request merged**